dd-sdk-android-core/src/main/kotlin/com/datadog/android/api/profiler/BenchmarkCounter.kt

```diff
@@ -0,0 +1,18 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.api.profiler
+
+import com.datadog.android.lint.InternalApi
+import com.datadog.tools.annotation.NoOpImplementation
+
+/**
+ * Interface of benchmark counter to be implemented to perform count metrics.
+ * This should only used by internal benchmarking.
+ */
+@NoOpImplementation
+@InternalApi
```

Comment:
>Warning
Documentation for InternalApi notes: "Don't use this annotation on interfaces". Is this intentional? Same question for the other interfaces in the PR.

Reply:
The reason given by the documentation why it shouldn't be used on interface is that :
> because implementation will be flagged as internal as a whole
Actually what I want is exactly that all the implementations from this interface should be internal.

But in case of confusion of the documentation, I removed all the `@InternalApi` on interface, I will put them manually on all the implementations, nice catch here!